### PR TITLE
Add description around CS and Replication issue, adding several release notes

### PR DESCRIPTION
**Pull request by kuenishi**

### CODE REVIEW

File: RELEASE-NOTES.md [hunk 1]
```diff
@@ -1,25 +1,126 @@
...
+  [riak_cs/#1148](https://github.com/basho/riak_cs/pull/1148),
+  [riak_cs/#1149](https://github.com/basho/riak_cs/pull/1149).
+
+- Refactor and rename `riak_cs_gc_d`, which is garbage collection
+  daemon by splitting user-facing GC execution state machine and
+  working GC daemon
+  [riak_cs/#1144](https://github.com/basho/riak_cs/pull/1144).
+
+- Add orphan blocks scanner and collector. In case inconsistencies
+  between manifests and blocks, especially manifests deleted and
```
Comment (shino) [inline RELEASE-NOTES.md]: The "manifests" here is not ones in manifest buckets but gc bucket. It is better to express it for accuracy, isn't it?

File: RELEASE-NOTES.md [hunk 2]
```diff
@@ -1,25 +1,126 @@
 # Riak CS 2.1.0 Release Notes
 
+## General Information
+
+This is a new release including new garbage collection, metrics system
```
Comment (shino) [inline RELEASE-NOTES.md]: I like somewhat restrained expression :sweat_smile:

File: RELEASE-NOTES.md (hunk 1 above)
Comment (shino) [inline RELEASE-NOTES.md]: I was wrong. `block_audit` script compares blocks and manifests in manifest buckets. Sorry.

File: RELEASE-NOTES.md [hunk 3]
```diff
@@ -1,25 +1,126 @@
...
+  daemon by splitting user-facing GC execution state machine and
+  working GC daemon
+  [riak_cs/#1144](https://github.com/basho/riak_cs/pull/1144).
+
+- Add orphan blocks scanner and collector. In case inconsistencies
+  between manifests and blocks, especially manifests deleted and
+  blocks remaining, there are no other way to collect those
+  blocks. The scanner compares full list of existing blocks and full
+  list of manifests under single bucket and figures out missing
+  manifests. THe collector deletes all such blocks to reclaim disk and
```
Comment (shino) [inline RELEASE-NOTES.md]: s/THe/The/

File: RELEASE-NOTES.md [hunk 4]
```diff
@@ -1,25 +1,126 @@
...
+concurrently and working on same blocks and manifests. Operators can
+specify the range of time period that garbage collector works on, to
+collect deleted objects synchronously on both sink and source sides.
+
+For Riak CS 2.0 or older, the way to avoid this issue is as follows.
+
+Set `{delete_mode, keep}` at `riak_kv` section in `advanced.config` of
+Riak. Instead this makes all tombstones remain forever in Riak, keys
+of all deleted blocks remain in cluster consuming fair amount of
+memory. Then blocks brought by fullsync will ignored according to the
```
Comment (shino) [inline RELEASE-NOTES.md]: s/ignored/be ignored/

**Comment by shino** [comment]
`select_gc_bucket` is another fancy feature of 2.1.

File: RELEASE-NOTES.md [hunk 5]
```diff
@@ -1,25 +1,126 @@
...
+This command means, garbage collector scans time slot starting from
+`epoch_start` defined in `advanced.config` (or its default `<<"0">>`)
+and deletes all blocks and manifests deleted in that time slot. After
+this garbage collection finished in both sink and source sides, The
+state of all manifests and blocks to be deleted are totally
+aligned. With this state fullsync replication may run without any
+inconsistency.
+
+This command is also useful to run garbage collection in incremental
+try-and-error manner by specifying start key of garbage collection
```
Comment (shino) [inline RELEASE-NOTES.md]: trial-and-error ?

**Comment by shino** [comment]
Thanks for writing up :clap: I added some small comments.

**Comment by kuenishi** [comment]
`select_gc_bucket.erl` is sort of an internal tool to know dominant disk usage. If we make this officially available and written in release notes, it should be some CUI that is to be installed in `/usr/bin`. Or do we have to do that for `block_audit` toolchain?

**Comment by kuenishi** [comment]
Also, updated to other parts of your comments.

File: RELEASE-NOTES.md [hunk 6]
```diff
@@ -1,25 +1,126 @@
 # Riak CS 2.1.0 Release Notes
 
+## General Information
+
+This release includes several garbage collection refinments, metrics
```
Comment (shino) [inline RELEASE-NOTES.md]: refinements

File: RELEASE-NOTES.md [hunk 7]
```diff
@@ -1,25 +1,126 @@
...
   whether they are halfway uploaded, or halfway deleted, or not.  This
-  is still off by default and
-  **experimental**. [riak_cs/#1120](https://github.com/basho/riak_cs/pull/1120),
+  is still off by default, but can be turned on by setting
+  `detailed_storage_calc` in
+  `advanced.config`. [riak_cs/#1120](https://github.com/basho/riak_cs/pull/1120),
   [riak_cs/#1123](https://github.com/basho/riak_cs/pull/1123),
 
+- Module-level hook point for limiting user access and quota usage has
+  been introduced, with very preliminary simple node-wide limting
```
Comment (shino) [inline RELEASE-NOTES.md]: limiting

**Comment by shino** [comment]
> If we make this officially available and written in release notes, it should be some CUI that is to be installed in /usr/bin. Or do we have to do that for block_audit toolchain?

Release notes had some preview/not-GA features in the past.  It's good to know what is introduced in what version from release notes.
On the other hand, because these three tools should be used with the assistance of customer support, they are not officially available (in other words, freely used).

Choices
1. Write the purposes and detailed description of these tools in release notes
2. Write the purposes shortly in release notes and add detailed description to the README in `priv/tools/`
3. Mention nothing

**Comment by kuenishi** [comment]
I chose 2. Typos fixed.

File: priv/tools/internal/ensure_orphan_blocks.erl [hunk 8]
```diff
@@ -46,7 +46,7 @@ main(Args) ->
     debug("Options: ~p", [Options]),
     case proplists:get_value(host, Options) of
         undefined ->
-            getopt:usage(option_spec(), "riak-cs escript /path/to/block_audit2.erl"),
+            getopt:usage(option_spec(), "riak-cs escript /path/to/ensure_orphan_blocks.erl"),
```
Comment (shino) [inline priv/tools/internal/ensure_orphan_blocks.erl:49]: Good catch :sweat_smile:

**Comment by kuenishi** [comment]
@borshop merge

**Comment by Basho-JIRA** [comment]
See https://github.com/basho/riak_cs/blob/develop/RELEASE-NOTES.md#new-riak-cs-admin-gc-command for release note. 

**_[posted via JIRA by Kota Uenishi]**_

**Pull request merged**